rlpx/handshake.go

```diff
@@ -3,7 +3,7 @@ package rlpx
 import (
 	"crypto/rand"
 	"errors"
-	mrand "math/rand"
+	// mrand "math/rand"
```

Comment:
Commenting this out for the time being since RLP can't seem to decode padded messages.

Reply:
I'm not so sure about this. I just tested the RLP package and I wasn't able to reproduce. Decode will return an error if the length of the input doesn't match the length encoded in the body. I also successfully decoded `rlp.List(rlp.Bytes([]byte{0x01, 0x00, 0x00}))`

Is this PR in a state where I can reproduce the bug? I un-commented out the padding code on this branch, but I got a seemingly unrelated error. Is that error in fact what you are referring to?